src/test/java/com/vinyl/ChangeOrderStatusTest.java

```diff
@@ -0,0 +1,88 @@
+package com.vinyl;
+
+import com.vinyl.helper.TokenHeaderHelper;
+import com.vinyl.model.Product;
+import com.vinyl.model.Purchase;
+import com.vinyl.model.PurchaseProduct;
+import com.vinyl.modelDTO.StatusDTO;
+import com.vinyl.repository.PurchaseRepository;
+import org.assertj.core.api.Assertions;
+import org.assertj.core.util.Lists;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.*;
+
+public class ChangeOrderStatusTest extends ManagerBaseIntegration {
+
+    @Autowired
+    private TokenHeaderHelper tokenHeaderHelper;
+
+    @Autowired
+    private PurchaseRepository purchaseRepository;
+
+    private Purchase order;
+
+    private Product product;
+
+    private StatusDTO status;
+
+    @Override
+    public void setUp() {
+        super.setUp();
+        product = defaultEntitiesHelper.createProduct();
+        order = purchaseSetup();
+        status = new StatusDTO();
+        status.setStatus("SENT");
+    }
+
+    @Test
+    public void testWhenOk() {
+        ResponseEntity<?> response = setUpHeaderAndGetTheResponse();
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+    }
+
+    @Test
+    public void testWhenStatusIsInvalid(){
+        status.setStatus("INVALID_STATUS");
+        ResponseEntity<?> response = setUpHeaderAndGetTheResponse();
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.BAD_REQUEST);
```

Comment:
assert that purchase status has not changed

Reply:
done